contracts/MyMultiSig.cdc

```diff
@@ -23,20 +12,68 @@ pub contract MyMultiSigV3 {
         access(contract) let multiSignManager: @Manager
     }
 
+    pub resource interface ManagerPublic {
+        pub fun borrowAction(actionUUID: UInt64): &MultiSignAction
+        pub fun getIDs(): [UInt64]
+        pub fun getIntents(): {UInt64: String}
+        pub fun getSigners(): {Address: Bool}
+        pub fun getThreshold(): UInt
+        pub fun getSignerResponsesForAction(actionUUID: UInt64): {Address: UInt}
+        pub fun getTotalApprovedForAction(actionUUID: UInt64): Int
+        pub fun getTotalRejectedForAction(actionUUID: UInt64): Int
+    }
+
     //
-    // ------- Action Wrapper ------- 
+    // ------- Struct Interfaces ------- 
     //
 
+    // Interface for actions implemented in TreasuryActions contract
     pub struct interface Action {
         pub let intent: String
         pub let proposer: Address
+        pub fun getView(): ActionView
         access(account) fun execute(_ params: {String: AnyStruct})
     }
 
     //
-    // ------- Structs --------
+    // ------- Structs ------- 
     //
 
+    // General View Struct for Actions
+    pub struct ActionView {
+        pub var name: String
+        pub var type: String
+        pub var intent: String
+        pub var proposer: Address
+        pub(set) var recipient: Address?
+        pub(set) var vaultId: String?
+        pub(set) var collectionId: String?
+        pub(set) var nftId: UInt64?
+        pub(set) var tokenAmount: UFix64?
+        pub(set) var signerAddr: Address?
+        pub(set) var newThreshold: UInt?
```

Comment:
What do you think about dividing this into multiple structs that inherit from ActionView?

Reply:
I was thinking about doing it that way, but decided to go with one single struct because its less code

Reply:
we are combining all the metadata into one struct, however I don't think there is any benefit of breaking them out into multiple structs.  the frontend will need to maintain a map of all actions and the params for each action anyway in order to display them.  so breaking out into separate structs doesn't actually provide any additional functionality or readability, unless i'm missing something.  i think the code is a bit more concise this way.